gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,127 @@
+The number of oracle operators is strictly limited to five, ensuring a balance between decentralization and manageability.
+
+## 3. Responsibility and Integrity
+
+Oracle operators are expected to prioritize the network's integrity and best interests, targeting an uptime of 99.9%. This goal serves as a benchmark to ensure high operational performance and accurate subgraph assessments. Recognizing the challenges of maintaining the targeted uptime, this standard does account for unforeseen issues beyond the operator's control, such as IPFS node or RPC service failures.
+
+To promote transparency, operators must provide a public dashboard displaying operational runs and failures, facilitating collective performance evaluation. Should an operator consistently underperform relative to its peers, the technical advisory board or other Subgraph Availability Oracle operators may suggest to the Graph Council to find a replacement.
+
+## 4. Subgraph Availability Oracle Software and Versioning
+
+Subgraph Availability Oracles operators must use the software available at [The Graph's Subgraph Oracle repository](https://github.com/graphprotocol/subgraph-oracle). This ensures uniformity and accuracy in their assessments across the network. A specific commit or release version from this repository will be designated as the required version for all oracles. This version is subject to change over time as updates and improvements are made.
+
+Oracle operators are responsible for reaching consensus on the specific commit and configuration to be used, and must align their operations with these agreed-upon standards by posting a transaction with the details to a `DataEdge` smart contract. A dedicated subgraph will track these declarations, flagging any deviations for prompt resolution to maintain network consistency.
+
+The transaction to the `DataEdge` should follow the format:
+
+```json
+{
+    "commit-hash": "XXXXX",
+    "config": {
+        "ipfs": "<IPFS_NODE_URL>",
+        "ipfs-concurrency": "4",
+        "ipfs-timeout": "10000",
+        "min-signal": "100",
+        "period": "300",
+        "grace_period": "0",
+        "supported-networks": "mainnet,gnosis,xdai,eip155:1,eip155:100,avalanche,eip155:43114,arbitrum-one,eip155:42161,celo,eip155:42220,fantom,eip155:250,matic,eip155:137",
+        "supported_data_source_kinds": "ethereum,ethereum/contract,file/ipfs,substreams,file/arweave",
+        "subgraph": "https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-arbitrum",
+        "subgraph_availability_manager_contract": "CONTRACT_ADDRESS",
+        "oracle_index": "ORACLE_INDEX",
+        "url": "<RPC_URL>"
+    }
+}
+
+```
+
+It is strongly recommended that Oracle operators diversify their sources for `IPFS_NODE_URL` and `RPC_URL` configurations. Utilizing varied sources for these critical components not only enhances the robustness and reliability of the network but also mitigates potential risks associated with single points of failure.
```

Comment:
Should there be any descriptions on operator's version upgrade process? especially when new versions become available, how to reach consensus among the oracle operators

Reply:
The intention is so that operators are responsible to coordinate between themselves to reach consensus for versioning and configurations. I believe this process will be clearer once we start working together but I'm happy to hear suggestions on how that might look like and if we want to include that process on the GIP. In my opinion it could be something that changes from time to time while we get more experience so I like having the GIP be flexible.

Reply:
Cool, I agree it would be good to keep the GIP flexible. 

Perhaps some options could be provided within the GIP? Like if the coordination process should be documented somewhere (github or notion or onchain?), if it should be internal or public, how long should it take to migrate between versions, ... general considerations like so.

Reply:
I'll update the GIP specifying that a group chat will be used as a communication channel and specify how long oracles have to update versions.